### ISSUE TITLE
test_spatial_cv_is_deterministic_under_a_seed is not deterministic

### ISSUE THREAD

**Issue by gcol33** [gcol33/resolve#115]
## Summary

`tests/core/test_spatial.py::test_spatial_cv_is_deterministic_under_a_seed` fails intermittently in CI. It asserts that two `cross_validate_spatial(block_config, 3, 7)` runs, seeded identically, pick the same `best_epoch` per fold.

Observed on run [33311089440](https://github.com/gcol33/resolve/actions/runs/33311089440) (commit 14793ee):

```
tests/core/test_spatial.py:66: in test_spatial_cv_is_deterministic_under_a_seed
    assert a.best_epoch == b.best_epoch
E   assert 0 == 1
```

## It is nondeterminism, not a regression

Same commit, same run: it failed on `ubuntu-latest / 3.12` and passed on `ubuntu-latest / 3.11`, `windows-latest / 3.11` and `windows-latest / 3.12`. Re-running the failed job on the identical commit passed. It also passes 5/5 locally on Windows. So the run-to-run outcome is not a function of the code under test.

## What is worth checking

`best_epoch` is chosen by comparing per-epoch test losses, so a flip between epoch 0 and epoch 1 means two epochs scored within float noise of each other AND that the two seeded runs did not produce bit-identical losses. The second half is the interesting one: a seeded run is supposed to be reproducible, and the engine already treats that as a contract elsewhere (`PretrainRng`, issue #107; the `--seed` reproducibility assertion in `cli-e2e`).

The likeliest cause is reduction order in a multi-threaded host op varying between runs, which would make this machine- and thread-count-dependent rather than seed-dependent. `at::set_num_threads(1)` around the test would confirm or rule that out.

## Two separable outcomes

1. If seeded runs really are not bit-reproducible on multiple threads, that is an engine-level finding worth documenting (and it bounds what `--seed` promises).
2. If they are, and only this tie-break is fragile, the assertion should compare the fold metrics within a tolerance, or the per-epoch loss histories, rather than an argmin over near-equal values.

Either way the test should stop failing at random: it is currently the only thing standing between a green `Tests` run and a red one on unchanged code.

## Not a blocker for v0.8.1

Found while verifying #111 / #112 / #113; none of those touch cross-validation, the RNG, or the training loop.

**Comment by gcol33** [comment]
Fixed on master in 3c484d7 — and it turned out not to be reduction order.

**Measured.** Six seeded `cross_validate_spatial` runs give six different fold losses at **one** thread as well as twenty-four, so threading was not it. Seeding the process-global torch RNG before constructing the model makes five runs bit-identical:

```
no torch.manual_seed:            mean mae = 1.834527, 1.851669, 1.829666, 1.843025, 1.817614
                                 best_epoch = (1,1,1) x4 and (1,1,0) once   <- the CI failure
torch.manual_seed(0) each time:  mean mae = 1.841990 x5, best_epoch = (1,1,1) x5
```

**Cause.** The seed passed to `prepare_data` and to `cross_validate_spatial` governs the **split**. Model weight initialisation draws from the process-global torch RNG, as any PyTorch module does, and nothing on the library path seeds it. The CLI already calls `torch::manual_seed` before constructing the model (that was #104); the library leaves it to the caller. The test asserted a determinism the API does not offer, and got it only when the noise happened not to move `best_epoch` across two epochs.

**What was not changed, deliberately.** Seeding a global RNG stream from inside `fit()` is exactly the side effect #107 avoided for pretraining, where the fix was a dedicated generator rather than touching the global stream. Weight init following the global RNG is also the ordinary PyTorch contract. So the engine is unchanged and the contract is written down instead (`docs/api/trainer.md` now shows the two-seed form).

**Pinned from both sides**, so this cannot come back as a flake:
- `test_spatial_cv_is_deterministic_under_a_seed` seeds split *and* weight init, and compares the fold **metrics**, not an argmin over them — a real divergence now shows as a value mismatch rather than hiding behind a tie-break.
- `test_an_unseeded_run_is_not_reproducible` asserts the gap is still there, and says in its failure message that if initialisation ever becomes seeded on the library path, it and the docs are stale.

Stable over 6 consecutive runs locally; full `tests/core` suite 134 passed.
